### PR TITLE
feat: add markuplint support

### PR DESCRIPTION
**Pull request by fang2hou**
# Changes

1. Add `markuplint` support.
   - homepage: https://markuplint.dev/
   - CLI reference: https://markuplint.dev/docs/guides/cli
2. Update readme.

# Screenshot

Screenshot 2024-02-24 at 0 02 48

### CODE REVIEW

File: lua/lint/linters/markuplint.lua [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if not ok then
+      return {
+        {
+          bufnr = bufnr,
+          lnum = 0,
+          col = 0,
+          message = "Could not parse markuplint output due to: " .. data .. "\noutput: " .. output,
+        },
+      }
+    end
```
Comment (mfussenegger) [inline lua/lint/linters/markuplint.lua:33]:
```suggestion
    local data = vim.json.decode(output, decode_opts)
```

This kind of error handling has become default and is no longer necessary per linter.
See https://github.com/mfussenegger/nvim-lint/blob/6f6d866362c74ec2eb036c32b56fac50b02af669/lua/lint/parser.lua#L143-L155

File: README.md [hunk 2]
```diff
@@ -137,6 +137,7 @@ Other dedicated linters that are built-in are:
 | [Languagetool][5]                  | `languagetool`         |
 | [luacheck][19]                     | `luacheck`             |
 | [markdownlint][26]                 | `markdownlint`         |
+| [markuplint][markuplint]                   | `markuplint`           |
 | [mlint][34]                        | `mlint`                |
```
Comment (mfussenegger) [inline README.md:141]:
```suggestion
| [markdownlint][26]                 | `markdownlint`         |
| [markuplint][markuplint]           | `markuplint`           |
| [mlint][34]                        | `mlint`                |
```

**Comment by fang2hou** [comment]
@mfussenegger 
Thank you for your advice!
And I also tested the code with the suggested commit applied, there is no issues.

**Comment by mfussenegger** [comment]
thanks

**Pull request merged**